controllers/machine/machine_controller.go

```diff
@@ -0,0 +1,194 @@
+		return r.evaluateDeletingMachine(ctx, machine)
+	}
+	return utils.DoNotRequeue()
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	indexerFunc := func(rawObj client.Object) []string {
+		event := rawObj.(*corev1.Event)
+		return []string{event.InvolvedObject.Name}
+	}
+
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Event{}, "involvedObject.name", indexerFunc); err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&machinev1beta1.Machine{}).
+		Complete(r)
+}
+
+func getMostRecentDrainFailedEvent(reqLogger logr.Logger, eventList *corev1.EventList) *corev1.Event {
+	// if there are no events at all exit now
+	if len(eventList.Items) == 0 {
+		reqLogger.Info("No events")
+		return nil
+	}
+
+	var newestEvent *corev1.Event
+
+	for i := range eventList.Items {
+		event := &eventList.Items[i]
+		if event.Reason == "DrainRequeued" {
+			if strings.Contains(event.Message, "error when evicting pods") {
+				if newestEvent == nil {
+					newestEvent = event
+					continue
+				}
+				if newestEvent.LastTimestamp.Time.Before(event.LastTimestamp.Time) {
+					newestEvent = event
+					continue
+				}
+			}
+		}
+	}
+
+	return newestEvent
+}
+
+func parsePodsAndNamespacesFromEvent(reqLogger logr.Logger, event *corev1.Event) map[string]string {
+	re := regexp.MustCompile(`pods\/"([\w-]+)" -n "([\w-]+)"`)
+	matches := re.FindAllStringSubmatch(event.Message, -1)
+
+	// store the pod/namespace combinations with the pod as the key because
+	// the pod name _should_ generally be unique, where there may be multiple pods
+	// in each namespace
+	podNamespaces := map[string]string{}
+
+	for _, podMatch := range matches {
+		if len(podMatch) != 3 {
+			// I don't think this should ever happen, but this prevents trying to access indexes
+			// in the match slice that may not exist
+			reqLogger.Error(fmt.Errorf("Could not get the appropriate amount of matches"), "match", podMatch)
+			continue
+		}
+		// From the regex match we'll always get this podMatch slice with the following format:
+		// ['pods/"myPod-aaabbb" -n "namespace"', 'myPod-aaabbb', 'namespace']
+		podName := podMatch[1]
+		podNamespace := podMatch[2]
+
+		namespaceRe := regexp.MustCompile(`openshift-.*`)
+		// We don't generally care about openshift namespaces for customer metrics
+		if !namespaceRe.MatchString(podNamespace) {
+			podNamespaces[podName] = podNamespace
+		}
+	}
+	return podNamespaces
+}
+
+func (r *MachineReconciler) evaluateDeletingMachine(ctx context.Context, machine *machinev1beta1.Machine) (ctrl.Result, error) {
+	reqLogger := logf.FromContext(ctx).WithName(logName)
+
+	// Check Deleting Timestamp. If it's been less than 15m we don't care, requeue for 5m.
+	deletedTime := machine.GetDeletionTimestamp().Time
+	now := time.Now()
+	if !deletedTime.Before(now.Add(-15 * time.Minute)) {
```

Comment:
Can you make the 15 minutes a Constant so that it's easier to adjust in the future?

Reply:
Done.